internal/io/sftpclient/sftpclient.go

```diff
@@ -57,57 +61,63 @@ func NewClient(source string) Client {
 	if p := sshCreds["port"]; p != "" {
 		port, err = strconv.Atoi(p)
 		if err != nil || port < 1 || port > 65535 {
-			logrus.Fatalf("Port number (%s) is wrong.", p)
+			err = errors.New("Port number " + p + " is wrong.")
+			return Client{}, err
```

Comment:
Since you aren't using the err for anything else I would not reassign. Just return the new error:
```golang
return Client{}, errors.New("Port number " + p + " is wrong.")
```